### PR TITLE
Add: Itinerary page

### LINKED ISSUE #136: Rider itinerary page (basic)
A Rider should be able to see all of the deliveries they are signed up for.

Right now we have "Delivery details pages" that work for a single delivery. 

We would like an "itinerary" that gives you a list of these ordered by day.

The "basic" version can just aggregate data (e.g. "You are delivering 10 items for 5 campaigns"), and link to the existing iteneraries.

Also note that we won't show the rides later this week to people because of how we dispatch - so we want some "Beta" copy.

### PR DESCRIPTION
**Pull request by teesloane**
I thought there would be more to do in building the itinerary page, but I think a lot of it was already done. 

This PR adds:

- "Itinerary" to the sidebar for riders
- remove ability to link to the campaign directly, since that's unauthorized for riders.
- on rider logging in, redirect them to the `/itinerary` route. 

If I understand correctly, this maybe closes #136? 

-----

Itinerary _ Bike Brigade Dispatch

### CODE REVIEW

File: lib/bike_brigade_web/live/itinerary_live/index.html.heex [hunk 1]
```diff
@@ -45,57 +45,51 @@
...
-            <div class="items-center justify-between md:flex">
-              <div class="flex items-center mb-2 space-x-1">
-                <.link
-                  navigate={~p"/campaigns/#{cr.campaign}"}
-                  class="font-bold link"
-                >
-                  <%= name(cr.campaign) %>
-                </.link>
+  <div class="flex flex-col mt-8">
+    <%= for cr <- @campaign_riders do %>
```
Comment (mveytsman) [inline lib/bike_brigade_web/live/itinerary_live/index.html.heex:49]: Might as well switch to the new heex syntax while we're here

**Review by mveytsman — CHANGES_REQUESTED** [review]
One small nit: let's switch to the `:for` [syntax](https://hexdocs.pm/phoenix_live_view/Phoenix.Component.html#sigil_H/2-heex-extension-special-attributes) as we work on the templates

File: lib/bike_brigade_web/live/itinerary_live/index.html.heex (hunk 1 above)
Comment (teesloane) [reply lib/bike_brigade_web/live/itinerary_live/index.html.heex]: Welp, that shrunk the PR 🥳

**Comment by teesloane** [comment]
> One small nit: let's switch to the `:for` [syntax](https://hexdocs.pm/phoenix_live_view/Phoenix.Component.html#sigil_H/2-heex-extension-special-attributes) as we work on the templates

Ready for re-review!

**Review by mveytsman — APPROVED** [review]
👍

**Pull request merged**